packages/cli/generation/ir-migrations/src/migrations/v29-to-v28/migrateFromV29ToV28.ts

```diff
@@ -55,11 +55,14 @@ function getV28TypeDeclarationFromId({
 }: {
     typeId: commons.TypeId;
     allTypes: Record<FernIr.TypeId, FernIr.TypeDeclaration>;
-}): FernIrV28.TypeDeclaration {
+}): IrVersions.V28.types.TypeDeclaration {
     const typeDeclaration = getTypeDeclarationOrThrow({ typeId, allTypes });
-
     return {
-        ...typeDeclaration,
+        availability: typeDeclaration.availability,
+        docs: typeDeclaration.docs,
+        name: typeDeclaration.name,
+        shape: typeDeclaration.shape,
+        examples: typeDeclaration.examples,
```

Comment:
So the initial code worked but was the wrong type? If you have time can you leave a brief explanation of this change? Thanks :)

Reply:
It was right, but i'm running into some weird issues with set -> list serialization 


```ts
types -> type_:GenerateRequestTruncate -> referencedTypes: Expected Set. Received list.; types -> type_:GenerateRequestReturnLikelihoods -> referencedTypes: Expected Set. Received list.; types -> type_:EmbedRequestTruncate -> referencedTypes: Expected Set. Received list.; types -> type_:EmbedResponse -> referencedTypes: Expected Set. Received list.; types -> type_:ClassifyRequestExamplesItem -> referencedTypes: Expected Set. Received list.; types -> type_:ClassifyRequestTruncate -> referencedTypes: Expected Set. Received list.; types -> type_:ClassifyResponse -> referencedTypes: Expected Set. Received list.; types -> type_:ClassifyResponseClassificationsItem -> referencedTypes: Expected Set. Received list.; types -> type_:ClassifyResponseClassificationsItemLabelsValue -> referencedTypes: Expected Set. Received list.; types -> type_:ClassifyResponseClassificationsItemClassificationType -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamRequestPromptTruncation -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamRequestCitationQuality -> referencedTypes: Expected Set. Received list.; types -> type_:ChatRequestPromptTruncation -> referencedTypes: Expected Set. Received list.; types -> type_:ChatRequestCitationQuality -> referencedTypes: Expected Set. Received list.; types -> type_:TokenizeResponse -> referencedTypes: Expected Set. Received list.; types -> type_:DetokenizeResponse -> referencedTypes: Expected Set. Received list.; types -> type_:DetectLanguageResponse -> referencedTypes: Expected Set. Received list.; types -> type_:DetectLanguageResponseResultsItem -> referencedTypes: Expected Set. Received list.; types -> type_:SummarizeRequestLength -> referencedTypes: Expected Set. Received list.; types -> type_:SummarizeRequestFormat -> referencedTypes: Expected Set. Received list.; types -> type_:SummarizeRequestExtractiveness -> referencedTypes: Expected Set. Received list.; types -> type_:SummarizeResponse -> referencedTypes: Expected Set. Received list.; types -> type_:SummarizeResponseResultsItem -> referencedTypes: Expected Set. Received list.; types -> type_:RerankRequestDocumentsItem -> referencedTypes: Expected Set. Received list.; types -> type_:RerankResponse -> referencedTypes: Expected Set. Received list.; types -> type_:RerankResponseResultsItem -> referencedTypes: Expected Set. Received list.; types -> type_:RerankResponseResultsItemDocument -> referencedTypes: Expected Set. Received list.; types -> type_:SingleGeneration -> referencedTypes: Expected Set. Received list.; types -> type_:SingleGenerationTokenLikelihoodsItem -> referencedTypes: Expected Set. Received list.; types -> type_:ApiMeta -> referencedTypes: Expected Set. Received list.; types -> type_:ApiMetaApiVersion -> referencedTypes: Expected Set. Received list.; types -> type_:Generation -> referencedTypes: Expected Set. Received list.; types -> type_:GenerationStream -> referencedTypes: Expected Set. Received list.; types -> type_:FinishReason -> referencedTypes: Expected Set. Received list.; types -> type_:SingleGenerationInStream -> referencedTypes: Expected Set. Received list.; types -> type_:GenerationFinalResponse -> referencedTypes: Expected Set. Received list.; types -> type_:GenerationFinalResponseResponse -> referencedTypes: Expected Set. Received list.; types -> type_:StreamedGeneration -> referencedTypes: Expected Set. Received list.; types -> type_:StreamedGenerationItem -> referencedTypes: Expected Set. Received list.; types -> type_:ChatMessage -> referencedTypes: Expected Set. Received list.; types -> type_:ChatMessageRole -> referencedTypes: Expected Set. Received list.; types -> type_:ChatConnector -> referencedTypes: Expected Set. Received list.; types -> type_:ChatDocument -> referencedTypes: Expected Set. Received list.; types -> type_:ChatCitation -> referencedTypes: Expected Set. Received list.; types -> type_:ChatSearchQuery -> referencedTypes: Expected Set. Received list.; types -> type_:ChatSearchResult -> referencedTypes: Expected Set. Received list.; types -> type_:NonStreamedChatResponse -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamEvent -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamStartEvent -> referencedTypes: Expected Set. Received list.; types -> type_:ChatSearchQueriesGenerationEvent -> referencedTypes: Expected Set. Received list.; types -> type_:ChatSearchResultsEvent -> referencedTypes: Expected Set. Received list.; types -> type_:ChatTextGenerationEvent -> referencedTypes: Expected Set. Received list.; types -> type_:ChatCitationGenerationEvent -> referencedTypes: Expected Set. Received list.; types -> type_:SearchQueriesOnlyResponse -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamEndEvent -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamEndEventFinishReason -> referencedTypes: Expected Set. Received list.; types -> type_:ChatStreamEndEventResponse -> referencedTypes: Expected Set. Received list.; types -> type_:StreamedChatResponse -> referencedTypes: Expected Set. Received list.
dsinghvi@Deeps-MacBook-Pro cohere-api % 
```

Reply:
i just disabled response validation to unblock and was thinking the spread operator could have something to do with it, but it doesnt